### PR TITLE
[10] [Backend] Add HTTP service

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/flutter-survey/issues/10

## What happened

We need a service to handle an HTTP request

### CODE REVIEW

**Comment by sleepylee** [comment]
note for the title of the PR, I think it would be good to see the title prefix i.e [Backend] in this case too, so we can navigate through your list of PRs more efficient imo.

File: lib/services/http/http_service.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  });
+}
+
+class HttpServiceImpl implements HttpService {
+  final _dio = Dio();
+
+  HttpServiceImpl() {
+    _dio.interceptors.add(LogInterceptor(
+      requestBody: true,
+      responseBody: true,
```
Comment (sleepylee) [inline lib/services/http/http_service.dart:23]: should we just enable these on debug? 🦖

**Review by sleepylee — COMMENTED** [review]
I reviewed this PR and https://github.com/markgravity/flutter-survey/pull/43/files as well...I feel like it's not a wise choice to use dio directly and create your owner request/response handler. In android we are familiar with `retrofit` and in iOS you guys have `alamofire` - shouldn't you setup an HTTP service wrapper with a similar concept here for flutter?

**Comment by manh-t** [comment]
so @sleepylee do you mean that we should setup a service wrapper like [retrofit](https://pub.dev/packages/retrofit)? 🙏 It's nice to have

**Comment by sleepylee** [comment]
still pending for your answer, @markgravity 🙇

**Comment by markgravity** [comment]
> still pending for your answer, @markgravity 🙇

I think I wrapped `dio` around `HttpService` already? 🤔

**Review by manh-t — APPROVED** [review]

**Review by sleepylee — APPROVED** [review]

**Pull request merged**